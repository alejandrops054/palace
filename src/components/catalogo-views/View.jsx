/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React from "react";
import "antd/dist/antd.css";
import { Table } from "antd";
import {
  Col,
  Row,
  Card,
  CardBody,
  CardTitle,
  Input,
  Label,
  Button,
  ButtonGroup,
} from "reactstrap";

export default function View() {
  const columns = [
    {
      title: "Menu ",
      dataIndex: "menu",
      key: "menu",
      width: 40,
      render: () => (
        <Input type="select">
          <option>Menu</option>
        </Input>
      ),
    },

    {
      title: "Description",
      dataIndex: "description",
      key: "description",
      width: 50,
    },
  ];
  const data = [
    {
      menu: "146",
      description: "Member Report",
    },
  ];
  const columns2 = [
    {
      title: "User Class ",
      dataIndex: "userclass",
      key: "userclass",
      width: 40,
      render: () => (
        <Input type="select">
          <option>CLB_VLOS_BP</option>
          <option>CLB_VLOS_MPS</option>
          <option>CLB_VLOS_PL</option>
          <option>CLB_VLOS_BP</option>
        </Input>
      ),
    },

    {
      title: "Description",
      dataIndex: "description",
      key: "description",
      width: 100,
    },
  ];
  const data2 = [
    {
      userclass: "",
      description: "Usuario Vlo's de Sala Beach Palace",
    },
    {
      userclass: "",
      description: "Usuario Vlo's de Sala Beach Palace",
    },
    {
      userclass: "",
      description: "Usuario Vlo's de Sala Beach Palace",
    },
  ];
  return (
    <>
      <Card>
        <CardBody>
          <CardTitle>
            <h2>View</h2>
          </CardTitle>
          <br />
          <Row>
            <Col xs="3">
              <Label>Code</Label>
              <Input type="text" />
            </Col>
            <Col xs="3"></Col>
            <Col xs="3">
              <br />
              <Label check>
                <Input type="checkbox" /> Special
              </Label>
            </Col>
          </Row>
          <br />
          <Row>
            <Col xs="8">
              <Label>Description</Label>
              <Input type="text" />
            </Col>
          </Row>
          <br />
          <Row>
            <Col xs="3">
              <Label>System</Label>
              <Input type="text" />
            </Col>
            <Col xs="3">
              <br />
              <Input type="select">
                <option>Systema de Contratos</option>
                <option>2</option>
                <option>3</option>
                <option>4</option>
                <option>5</option>
              </Input>
            </Col>
          </Row>
          <br />
          <Row>
            <Col xs="5">
              <Label>Status Help</Label>
              <Input type="text" />
            </Col>
          </Row>
          <br />
          <Row>
            <Col xs="5">
              <Label>Tag</Label>
              <Input type="text" />
            </Col>
          </Row>
          <br />
          <Row>
            <Col xs="12">
              <Table columns={columns} dataSource={data} />
              <span className="owner">Menus : 1</span>
            </Col>
          </Row>
          <br />
          <Row>
            <Col xs="12">
              <Table columns={columns2} dataSource={data2} />
              <span className="owner">User Classes : 13</span>
            </Col>
          </Row>
          <Row>
            <Col xs="8"></Col>
            <Col xs="4">
              <ButtonGroup>
                <Button color="primary">Save</Button>
                <Button color="primary">Close</Button>
              </ButtonGroup>
            </Col>
          </Row>
        </CardBody>
      </Card>
    </>
  );
}
