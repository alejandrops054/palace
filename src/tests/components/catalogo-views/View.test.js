/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import View from '../../../components/catalogo-views/View'

describe('View component',() => {
    test('Debe mostrar snapshot', () => {
        
        const wrapper = shallow(<View/>)
        expect(wrapper).toMatchSnapshot();
    })
    
})