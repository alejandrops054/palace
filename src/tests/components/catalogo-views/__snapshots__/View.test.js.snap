// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component Debe mostrar snapshot 1`] = `
<Fragment>
  <Card
    tag="div"
  >
    <CardBody
      tag="div"
    >
      <CardTitle
        tag="div"
      >
        <h2>
          View
        </h2>
      </CardTitle>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="3"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Code
          </Label>
          <Input
            type="text"
          />
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="3"
        />
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="3"
        >
          <br />
          <Label
            check={true}
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Input
              type="checkbox"
            />
             Special
          </Label>
        </Col>
      </Row>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="8"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Description
          </Label>
          <Input
            type="text"
          />
        </Col>
      </Row>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="3"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            System
          </Label>
          <Input
            type="text"
          />
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="3"
        >
          <br />
          <Input
            type="select"
          >
            <option>
              Systema de Contratos
            </option>
            <option>
              2
            </option>
            <option>
              3
            </option>
            <option>
              4
            </option>
            <option>
              5
            </option>
          </Input>
        </Col>
      </Row>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="5"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Status Help
          </Label>
          <Input
            type="text"
          />
        </Col>
      </Row>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="5"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Tag
          </Label>
          <Input
            type="text"
          />
        </Col>
      </Row>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Table
            columns={
              Array [
                Object {
                  "dataIndex": "menu",
                  "key": "menu",
                  "render": [Function],
                  "title": "Menu ",
                  "width": 40,
                },
                Object {
                  "dataIndex": "description",
                  "key": "description",
                  "title": "Description",
                  "width": 50,
                },
              ]
            }
            dataSource={
              Array [
                Object {
                  "description": "Member Report",
                  "menu": "146",
                },
              ]
            }
            rowKey="key"
          />
          <span
            className="owner"
          >
            Menus : 1
          </span>
        </Col>
      </Row>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Table
            columns={
              Array [
                Object {
                  "dataIndex": "userclass",
                  "key": "userclass",
                  "render": [Function],
                  "title": "User Class ",
                  "width": 40,
                },
                Object {
                  "dataIndex": "description",
                  "key": "description",
                  "title": "Description",
                  "width": 100,
                },
              ]
            }
            dataSource={
              Array [
                Object {
                  "description": "Usuario Vlo's de Sala Beach Palace",
                  "userclass": "",
                },
                Object {
                  "description": "Usuario Vlo's de Sala Beach Palace",
                  "userclass": "",
                },
                Object {
                  "description": "Usuario Vlo's de Sala Beach Palace",
                  "userclass": "",
                },
              ]
            }
            rowKey="key"
          />
          <span
            className="owner"
          >
            User Classes : 13
          </span>
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="8"
        />
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="4"
        >
          <ButtonGroup
            role="group"
            tag="div"
          >
            <Button
              color="primary"
              tag="button"
            >
              Save
            </Button>
            <Button
              color="primary"
              tag="button"
            >
              Close
            </Button>
          </ButtonGroup>
        </Col>
      </Row>
    </CardBody>
  </Card>
</Fragment>
`;
